src/crimson/os/seastore/btree/fixed_kv_node.h

```diff
@@ -275,13 +275,22 @@ struct FixedKVNode : ChildableCachedExtent {
     node_key_t key)
   {
     assert(children.capacity());
-    auto child = children[pos];
+    ceph_assert(is_valid());
+    ChildableCachedExtent* child = nullptr;
+    if (key == get_key_from_idx(pos)) {
+      child = children[pos];
+    } else {
+      auto npos = lower_bound_offset(key);
+      assert(key == get_key_from_idx(npos));
+      child = children[npos];
```

Comment:
Seems to me this is a special case due to the usage during `ObjectDataHandler::overwrite()`. So, instead of handling it in all of the get_child pathes, is it better to introduce a special interface to fix the `pos` during `ObjectDataHandler::overwrite()` when necessary ?

The dedicated interface can also be a reminder for the special case.